python/cudf/dataframe/multiindex.py

```diff
@@ -465,5 +465,32 @@ def from_pandas(cls, multiindex):
                      names=multiindex.names)
         return mi
 
+    @property
+    def is_unique(self):
+        if not hasattr(self, '_is_unique'):
+            self._is_unique = (
+                self._source_data._size
+                == self._source_data.drop_duplicates()._size
+            )
+        return self._is_unique
+
+    @property
+    def is_monotonic_increasing(self):
+        if not hasattr(self, '_is_monotonic_increasing'):
+            values = [
+                self._source_data[i] for i in reversed(self._source_data._cols)
+            ]
+            sort_order = np.lexsort(values)
```

Comment:
Instead of using `np.lexsort` you should be able to use `cudf.DataFrame.argsort` to do the same as the lexsort here, except it will run on device 😄 and return you a NumericalColumn that you can check the monotonic properties of.
```
self._source_data.argsort().is_monotic_increasing
```

Reply:
Perfect, that’s exactly what I was looking for - Thanks!